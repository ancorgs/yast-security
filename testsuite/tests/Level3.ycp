/*
 * YaST2: Modules testsuite
 *
 * Description:
 *   Testsuite for the security module
 *
 * Authors:
 *   Michal Svec <msvec@suse.cz>
 *
 * $Id$
 */

{

include "testsuite.ycp";

map DATA = $[

      /* "Boot" */
      "CONSOLE_SHUTDOWN"	: "ignore",
      "KDM_SHUTDOWN"		: "root",
      /* "Login" */
      "FAILLOG_ENAB"		: "yes",
      "LASTLOG_ENAB"		: "yes",
      "FAIL_DELAY"		: "3",
      "ROOT_LOGIN_REMOTE"	: "no",
      "PASSWD_USE_CRACKLIB"	: "yes",
      "OBSCURE_CHECKS_ENAB"	: "no",
      "PASS_WARN_AGE"		: "14",
      "MD5_SUPPORT"		: "no",
      /* "Perms" */
      "PERMISSION_SECURITY"	: "secure",
      /* "Other" */
      "CWD_IN_ROOT_PATH"	: "no",
      "RUN_UPDATEDB_AS"		: "nobody",
      /* Adduser */
      "GID_MIN"			: "101",
      "GID_MAX"			: "60000",
      "UID_MIN"			: "100",
      "UID_MAX"			: "60000",
      "PASS_MIN_DAYS"		: "1",
      "PASS_MAX_DAYS"		: "99999",
      "PASS_MIN_LEN"		: "6",
      "PASS_MAX_LEN"		: "8"
];

TEST(`security_write(DATA),[],false);

}
