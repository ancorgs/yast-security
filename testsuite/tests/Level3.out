Read	.etc.login_defs.FAILLOG_ENAB nil
Write	.etc.login_defs.FAILLOG_ENAB "yes" true
Read	.etc.login_defs.FAIL_DELAY nil
Write	.etc.login_defs.FAIL_DELAY "3" true
Read	.etc.login_defs.GID_MAX nil
Write	.etc.login_defs.GID_MAX "60000" true
Read	.etc.login_defs.GID_MIN nil
Write	.etc.login_defs.GID_MIN "1000" true
Read	.etc.login_defs.LASTLOG_ENAB nil
Write	.etc.login_defs.LASTLOG_ENAB "yes" true
Read	.etc.login_defs.PASS_MAX_DAYS nil
Write	.etc.login_defs.PASS_MAX_DAYS "99999" true
Read	.etc.login_defs.PASS_MIN_DAYS nil
Write	.etc.login_defs.PASS_MIN_DAYS "1" true
Read	.etc.login_defs.PASS_WARN_AGE nil
Write	.etc.login_defs.PASS_WARN_AGE "14" true
Read	.etc.login_defs.UID_MAX nil
Write	.etc.login_defs.UID_MAX "60000" true
Read	.etc.login_defs.UID_MIN nil
Write	.etc.login_defs.UID_MIN "1000" true
Read	.etc.login_defs.SYSTEM_UID_MAX nil
Write	.etc.login_defs.SYSTEM_UID_MAX "499" true
Read	.etc.login_defs.SYSTEM_UID_MIN nil
Write	.etc.login_defs.SYSTEM_UID_MIN "100" true
Read	.etc.login_defs.SYSTEM_GID_MAX nil
Write	.etc.login_defs.SYSTEM_GID_MAX "499" true
Read	.etc.login_defs.SYSTEM_GID_MIN nil
Write	.etc.login_defs.SYSTEM_GID_MIN "100" true
Read	.etc.login_defs.USERADD_CMD nil
Write	.etc.login_defs.USERADD_CMD "/usr/sbin/useradd.local" true
Read	.etc.login_defs.USERDEL_PRECMD nil
Write	.etc.login_defs.USERDEL_PRECMD "/usr/sbin/userdel-pre.local" true
Read	.etc.login_defs.USERDEL_POSTCMD nil
Write	.etc.login_defs.USERDEL_POSTCMD "/usr/sbin/userdel-post.local" true
Read	.sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS nil
Write	.sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS "no" true
Read	.sysconfig.displaymanager.KDM_SHUTDOWN nil
Write	.sysconfig.displaymanager.KDM_SHUTDOWN "root" true
Read	.sysconfig.locate.RUN_UPDATEDB_AS nil
Write	.sysconfig.locate.RUN_UPDATEDB_AS "nobody" true
Read	.sysconfig.security.PERMISSION_SECURITY nil
Write	.sysconfig.security.PERMISSION_SECURITY "secure local" true
Read	.sysconfig.suseconfig.CWD_IN_ROOT_PATH nil
Write	.sysconfig.suseconfig.CWD_IN_ROOT_PATH "no" true
Read	.sysconfig.suseconfig.CWD_IN_USER_PATH nil
Write	.sysconfig.suseconfig.CWD_IN_USER_PATH "no" true
Read	.sysconfig.sysctl.ENABLE_SYSRQ nil
Write	.sysconfig.sysctl.ENABLE_SYSRQ "no" true
Write	.etc.login_defs nil true
Write	.sysconfig.displaymanager nil true
Write	.sysconfig.locate nil true
Write	.sysconfig.security nil true
Write	.sysconfig.suseconfig nil true
Write	.sysconfig.sysctl nil true
Write	.etc.inittab.ca ":ctrlaltdel:/bin/true" true
Write	.etc.inittab nil true
Read	.etc.default.passwd."CRYPT_FILES" nil
Write	.etc.default.passwd."CRYPT_FILES" "blowfish" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "cracklib" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "no_obscure_checks" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "" true
Dir	.etc.security.section: []
Read	.etc.security.v."pam_pwcheck"."password" nil
Dir	.etc.security.section: []
Write	.etc.security.v."pam_pwcheck"."password" "minlen=6" true
Write	.etc.security nil true
Write	.etc.default.passwd nil true
Execute	.target.bash "echo 0 > /proc/sys/kernel/sysrq" 0
Execute	.target.bash "/sbin/SuSEconfig --module xdm" 0
Execute	.target.bash "/sbin/SuSEconfig --module kde3" 0
Execute	.target.bash "/sbin/SuSEconfig --module kdm3" 0
Execute	.target.bash "/sbin/SuSEconfig --module permissions" 0
Execute	.target.bash "/sbin/SuSEconfig --module profiles" 0
Return	true
